### ISSUE TITLE
JS execution converts <object> elements to strings

### ISSUE THREAD

**Issue by rikva** [SeleniumHQ/selenium#3186]
## Meta -
OS:  Mac OS and Linux
Selenium Version:  2.53.6
Browser: Firefox
Browser Version: 47.0.2

## Expected Behavior -
I would expect <object> elements to be converted to `WebElement` objects just as other elements.

## Actual Behavior -
JS execution returns `WebElement` objects for all elements instead for <object> elements; they are returned as strings.

## Steps to reproduce -
```python
from selenium import webdriver

url = "file:///project_dir/object.html"

""" 
"object.html" contains:

<h1>example</h1>
<object width="400" height="400" data="helloworld.swf">
</object>
"""

b = webdriver.Firefox()
b.get(url)
js = 'return document.getElementsByTagName("*");'

for el in b.execute_script(js):
    print(type(el), el)
```

Output (note the last line)
```
<class 'selenium.webdriver.remote.webelement.WebElement'> <selenium.webdriver.remote.webelement.WebElement (session="f6b05e8f-1140-6942-b11d-4d7f6d7f57e0", element="{a23b1b1c-cc90-7043-ac8c-ac33a4b3d195}")>
<class 'selenium.webdriver.remote.webelement.WebElement'> <selenium.webdriver.remote.webelement.WebElement (session="f6b05e8f-1140-6942-b11d-4d7f6d7f57e0", element="{2f088222-c2e6-5b4e-aa38-f14fb6e98e55}")>
<class 'selenium.webdriver.remote.webelement.WebElement'> <selenium.webdriver.remote.webelement.WebElement (session="f6b05e8f-1140-6942-b11d-4d7f6d7f57e0", element="{b37b01cf-f13f-7943-8c22-ee3e997853d3}")>
<class 'selenium.webdriver.remote.webelement.WebElement'> <selenium.webdriver.remote.webelement.WebElement (session="f6b05e8f-1140-6942-b11d-4d7f6d7f57e0", element="{e90318ac-a1c8-b048-86b5-b0649c994571}")>
<class 'str'> [object HTMLObjectElement]
```

**Comment by rikva** [comment]
After trying to debug this issue, I'm fairly certain that the issue might not be in Selenium/WebDriver but in Firefox, instead. 

This is the raw response data from the request to the FF webdriver, the distinction between regular and `object` elements is already obvious.

(indentation is added for readability)
```python
b'{
"name":"executeScript",
"sessionId":"7f209e2a-8397-704b-9f95-a0ee958dd8d4",
"status":0,
"value": [
    {"ELEMENT":"{c9bd03c3-0631-8441-977a-d230ee316c23}"},
    {"ELEMENT":"{72b1d2a3-27f6-f64c-8d00-3d97116383d6}"},
    {"ELEMENT":"{fd1f2fe5-8ee1-e84a-9fba-46112739cd21}"},
    {"ELEMENT":"{f8c6a1ae-92c8-7447-9b80-70573b1bbaba}"},
    "[object HTMLObjectElement]"
]}' 
```

**Comment by rikva** [comment]
Hi barancev, thanks for looking into this ticket.

Sorry about the test case, I thought it was adequate.

Unfortunately I can't reproduce your results:
I've tried webdriver 2.53.6 and 3.0.2 as well as Python 2.7 and 3.5, with equal results.

![schermafbeelding 2016-12-13 om 11 17 19](https://cloud.githubusercontent.com/assets/1089611/21136383/c3183412-c125-11e6-8f56-d5359bfc8ddc.png)

**Comment by barancev** [comment]
Probably a Mac-specific issue

**Comment by rikva** [comment]
Same result when running (the same version, linux-x64 build of) Firefox in a Ubuntu docker container:

![schermafbeelding 2016-12-14 om 11 26 28](https://cloud.githubusercontent.com/assets/1089611/21178528/38dfa5ac-c1f0-11e6-977b-3a4792cb446e.png)

I guess it could be unix/posix specific...

**Comment by aneeshchandy** [comment]
I was not able to reproduce this issue in FF 50.0 in Mac (Yosemite) using Selenium 3.0.1 and python 2.7.10 (testing the w3schools link)

```
3.0.1
2.7.10 (default, Jul 14 2015, 19:46:27) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]
Darwin-14.5.0-x86_64-i386-64bit

(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d72c8cc0-004b-4f47-ba10-b83fb8f4bff9", element="a03e87d0-ba4a-2849-a208-9d7aa059a062")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d72c8cc0-004b-4f47-ba10-b83fb8f4bff9", element="ef45291b-acfd-a143-a423-18de312eb934")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d72c8cc0-004b-4f47-ba10-b83fb8f4bff9", element="312f65b1-5291-b04d-a999-cb1666a1c571")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d72c8cc0-004b-4f47-ba10-b83fb8f4bff9", element="83255ac5-12b1-de42-8872-931ddbb20628")>)
```

Testing the object.html I see the same result
```
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d549d375-e76b-e34a-8985-484263c37855", element="007bf426-cdce-b04b-b5d1-1121cae98b57")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d549d375-e76b-e34a-8985-484263c37855", element="49bece21-f6c5-384d-9036-1608f6002c80")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d549d375-e76b-e34a-8985-484263c37855", element="c61a35f7-e591-d843-94ef-daa9aa1d4061")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d549d375-e76b-e34a-8985-484263c37855", element="455d1588-1a3e-6c48-9839-8baa3d8599c3")>)
(<class 'selenium.webdriver.firefox.webelement.FirefoxWebElement'>, <selenium.webdriver.firefox.webelement.FirefoxWebElement (session="d549d375-e76b-e34a-8985-484263c37855", element="a1054057-2754-3f4d-9c4d-b80c7e69faad")>)
```

**Comment by cgoldberg** [comment]
@rikva 
It looks like this issue is was with the legacy Firefox driver.  Can the OP reproduce this issue running a newer Selenium and Firefox, along with geckodriver?

**Comment by rikva** [comment]
Wait, I feel I should clarify this:

**This issue only occurs when returning elements from a execute_script call.**

Allow me to demonstrate:
![schermafbeelding 2017-01-31 om 13 02 11](https://cloud.githubusercontent.com/assets/1089611/22464201/e8a2e9be-e7b5-11e6-8539-ce09424f104d.png)

**Comment by rikva** [comment]
However, I can confirm that this issue seems to be fixed in Firefox 51.0:

![schermafbeelding 2017-01-31 om 13 20 49](https://cloud.githubusercontent.com/assets/1089611/22464641/2f696f24-e7b8-11e6-9e59-af48f6e04fe9.png)

**Comment by barancev** [comment]
Development of XPI-based Firefox driver has been discontinued and this issue will never be fixed in it (even if it is still actual).
